src/main/kotlin/com/amazon/opendistroforelasticsearch/indexstatemanagement/action/ForceMergeAction.kt

```diff
@@ -0,0 +1,74 @@
+import com.amazon.opendistroforelasticsearch.indexstatemanagement.model.action.ForceMergeActionConfig
+import com.amazon.opendistroforelasticsearch.indexstatemanagement.step.Step
+import com.amazon.opendistroforelasticsearch.indexstatemanagement.step.forcemerge.AttemptCallForceMergeStep
+import com.amazon.opendistroforelasticsearch.indexstatemanagement.step.forcemerge.AttemptRevertReadOnlyStep
+import com.amazon.opendistroforelasticsearch.indexstatemanagement.step.forcemerge.AttemptSetReadOnlyStep
+import com.amazon.opendistroforelasticsearch.indexstatemanagement.step.forcemerge.WaitForForceMergeStep
+import org.elasticsearch.client.Client
+import org.elasticsearch.cluster.service.ClusterService
+
+class ForceMergeAction(
+    clusterService: ClusterService,
+    client: Client,
+    managedIndexMetaData: ManagedIndexMetaData,
+    config: ForceMergeActionConfig
+) : Action(ActionType.FORCE_MERGE, config, managedIndexMetaData) {
+
+    private val attemptSetReadOnlyStep = AttemptSetReadOnlyStep(clusterService, client, config, managedIndexMetaData)
+    private val attemptCallForceMergeStep = AttemptCallForceMergeStep(clusterService, client, config, managedIndexMetaData)
+    private val waitForForceMergeStep = WaitForForceMergeStep(clusterService, client, config, managedIndexMetaData)
+    private val attemptRevertReadOnlyStep = AttemptRevertReadOnlyStep(clusterService, client, config, managedIndexMetaData)
+
+    // Using a LinkedHashMap here to maintain order of steps for getSteps() while providing a convenient way to
+    // get the current Step object using the current step's name in getStepToExecute()
+    private val stepNameToStep: LinkedHashMap<String, Step> = linkedMapOf(
+        AttemptSetReadOnlyStep.name to attemptSetReadOnlyStep,
+        AttemptCallForceMergeStep.name to attemptCallForceMergeStep,
+        WaitForForceMergeStep.name to waitForForceMergeStep,
+        AttemptRevertReadOnlyStep.name to attemptRevertReadOnlyStep
+    )
+
+    override fun getSteps(): List<Step> = stepNameToStep.values.toList()
+
+    @Suppress("ReturnCount")
+    override fun getStepToExecute(): Step {
+        // If stepMetaData is null, return the first step in ForceMergeAction
+        val stepMetaData = managedIndexMetaData.stepMetaData ?: return attemptSetReadOnlyStep
+
+        val currentStep = stepMetaData.name
+        val currentStepStatus = stepMetaData.stepStatus
+
+        // If the current step has completed, return the next step
+        if (currentStepStatus == Step.StepStatus.COMPLETED) {
+            return when (currentStep) {
+                AttemptSetReadOnlyStep.name -> attemptCallForceMergeStep
+                AttemptCallForceMergeStep.name -> waitForForceMergeStep
+                WaitForForceMergeStep.name -> attemptRevertReadOnlyStep
+                // Shouldn't hit this case but including it so that the when expression is exhaustive
+                else -> stepNameToStep[currentStep]!!
```

Comment:
You can replace with `stepNameToStep[currentStep]` and check if the map returns object with non-null object. If the returned object is null then we can log the `currentStep` and throw an error.

Reply:
Replaced returning ```stepNameToStep[currentStep]!!``` with throwing an ```IllegalArgumentException``` for the ```else``` case.